addons/l10n_my_edi_extended/models/account_move.py

```diff
@@ -0,0 +1,188 @@
+    # Compute, inverse, search methods
+    # --------------------------------
+
+    @api.depends('move_type', 'state', 'country_code', 'company_id')
+    def _compute_l10n_my_invoice_need_edi(self):
+        for move in self:
+            # We return true for malaysian invoices which are not sent yet, sent but awaiting validation or valid.
+            move.l10n_my_invoice_need_edi = self.env['account.move.send']._l10n_my_edi_need_edi(move, ['in_progress', 'valid'])
+
+    def _get_name_invoice_report(self):
+        # EXTENDS 'account'
+        if self.l10n_my_edi_external_uuid:  # Meaning we are a myinvois invoice, meaning we need to embed the qr code.
+            # As we add the view in stable, we need to check that it exists.
+            if self.env.ref('l10n_my_edi_extended.report_invoice_document', raise_if_not_found=False):
+                return 'l10n_my_edi_extended.report_invoice_document'
+        return super()._get_name_invoice_report()
+
+    # --------------
+    # Action methods
+    # --------------
+
+    def action_invoice_sent(self):
+        """ The wizard should not be available for invoices sent to MyInvois but not yet validated.
+        This is because before validation the ID used for the QR code is not available and the user should NOT send the invoice yet.
+        """
+        self.ensure_one()
+
+        if self.l10n_my_edi_state == 'in_progress':
+            raise UserError(_('You cannot send invoices that are currently being validated.\nPlease wait for the validation to complete.'))
+
+        return super().action_invoice_sent()
+
+    # ----------------
+    # Business methods
+    # ----------------
+
+    def _update_validation_fields(self, validation_result):
+        """ Extended to update the long id as well. """
+        # EXTENDS 'l10n_my_edi'
+        super()._update_validation_fields(validation_result)
+        self.l10n_my_edi_invoice_long_id = validation_result['long_id']
+
+    def _generate_myinvois_qr_code(self):
+        """ Generate the qr code which should be embedded into the invoices PDF """
+        self.ensure_one()
+
+        if not self.l10n_my_edi_invoice_long_id:  # Only valid invoices have a long id
+            return None
+
+        # We need to add the portal url to the qr
+        proxy_user = self._l10n_my_edi_ensure_proxy_user()
+        if proxy_user.edi_mode == 'prod':
+            portal_url = "myinvois.hasil.gov.my"
+        else:
+            portal_url = "preprod.myinvois.hasil.gov.my"
+
+        try:
+            qr_code = self.env['ir.actions.report'].barcode(
+                barcode_type='QR',
+                width=128,
+                height=128,
+                humanreadable=1,
+                value=f'https://{portal_url}/{self.l10n_my_edi_external_uuid}/share/{self.l10n_my_edi_invoice_long_id}',
+            )
+        except (ValueError, AttributeError):
+            raise werkzeug.exceptions.HTTPException(description='Cannot convert into QR Code.')
```

Comment:
http?

Reply:
I followed what we do in res.partner.bank when we generate a payment QR code; I'm not an expert on qr code generation errors so I thought it was safer